### DISCUSSION TITLE
Use case: On battery 24/7

### DISCUSSION THREAD

**Discussion by rthorntn** [davidusb-geek/emhass#295] [Q&A]
Hi David,

I've got to say a big thank you for all the work you put in here, EMHASS is epic.

I'm trying to figure out if EMHASS makes sense for me.

System located in Sydney, Australia:

7.4kw panels, 19.2kw batteries, an average of about 16kw daily usage

Three different inverters (MPP, Growatt, Goodwe), panels by default go into MPP to charge the batteries and power the house load but can be DC switched to Growatt for solar export only (as the MPPs don't support export to grid).  Goodwe is for battery import/export to grid.

Victron battery monitor and 3EM CT system with clamps on grid, growatt & goodwe

5 minute utility billing

Currently I have all my data in mqtt, node-red, influxdb & grafana

So my system was built for energy security, I never really expected to break even and then wholesale cost 5m billing came along and so my strategy changed a bit.

I'd like to:

Have the house run on battery 24/7

Buy/sell power in 5 minute chunks when the price makes sense

Never let the battery go below a 25% SoC unless the no solar and the power price is bonkers.

Sunny summer days I don't have to buy power at all, ever.

Top up from grid on:
Cloudy summer days
Sunny winter days
Cloudy winter days

I don't really have a deferred load mentality as I'm on battery 24/7.  I could automate my homes heat-pump hot water system and pool pump if it made sense.

Ideally, every day I'd like to get to solar sunset with a 100% SoC (as cheaply as possible) and try and make some money selling battery when the price spikes, waking up in the morning to a 25% SoC.

So can EMHASS support this use case?

Thanks again!

**Comment by purcell-lab** [comment]
EMHASS is ideally suited to your household configuration, one of the key strengths of EMHASS is you can tune various parameters to make it optimal to your desired outcomes.  If you haven't found it already there is a lot of users experiences and solutions available https://community.home-assistant.io/t/emhass-an-energy-management-for-home-assistant/338126?u=markpurcell

I'm located in Qld have 18 kWp solar, 40 kWh houshold battery, 2x EV, pool filter and heat pump, air con and domestic hot water heat pump, my energy retailer is Amber, so I'm very familiar with optimisation for wholesale pricing highs of $15+/ kWh and lows of  -$0.10/ kWh. From your description it sounds like you might also be with Amber or more likely LocalVolts.

Here is what the next 24 hours looks like for me:
![image](https://github.com/davidusb-geek/emhass/assets/79175134/b9316447-5d86-4ab3-9779-04e0cee74405)


> Have the house run on battery 24/7

EMHASS has a self consumption mode or you can also switch on/ off consumption/ export to grid modes, which would schedule everything to be self contained.

> Buy/sell power in 5 minute chunks when the price makes sense

EMHASS has recently been updated to support 5 minute intervals for forecasts.  I run the high frequency MPC optimisation every minute which recalculates the optim for my batteries/ devices depending on the inputs.

> Never let the battery go below a 25% SoC unless the no solar and the power price is bonkers.

You can specify an overall minimum SOC and a final SOC for each optimisation run.  You can also place a weight penalty on charging/ discharging which has the effect of preventing the system from chasing small spreads between import/ export costs (say 1-2c) if you would like a threshold of say 40c/ kWh to only catch the price spikes.

> Sunny summer days I don't have to buy power at all, ever.
> 
> Top up from grid on: Cloudy summer days Sunny winter days Cloudy winter days

Depending on the strategy EMHASS has set it, will calculate the optimal time and power levels for grid consumption to either maximise profit or self consumption.  It works very well with Victron and goodwe for charging/ import and discharging/ export though standard Home Assistant integrations.

> I don't really have a deferred load mentality as I'm on battery 24/7. I could automate my homes heat-pump hot water system and pool pump if it made sense.

With the pricing spread on the wholesale market I do find it an advantage to automate other household loads to act as a sink to soak up extremely_low prices, but also switch everything down during price spike events to maximise exports.

> > Ideally, every day I'd like to get to solar sunset with a 100% SoC (as cheaply as possible) and try and make some money selling battery when the price spikes, waking up in the morning to a 25% SoC.

One of the key metrics I chase is exactly that, do I hit sunset with maximum SOC and do I hit sunrise with minimum SOC and I find the level of detail EMHASS can provide is a key enabler of this down to the W.  You can see my SOC for the next 24 hours in the middle left of my graphic.  In fact it is going to be a little cheeky tonight exporting to minimum SOC at 11pm for the evening peak, charging up on cheap prices overnight and then hitting another minimum SOC at 8am after the morning peak.

> So can emhass support this use case?

I'm pretty pumped with the level of control and sophistication provided by EMHASS, the setup can be complex, but I haven't been able to find anything that can optimise my household energy management to this high performance.

**Comment by rthorntn** [comment]
G'day Mark, thanks heaps, what an amazing response!

With HASS, is it essential or does it make life heaps easier?

I don't have HASS installed right now, just node-red, happy to deploy it... but if I don't need to...

Maybe the setup is all done in HASS, so it makes setting up EMHASS a lot easier?

**Comment by davidusb-geek** [comment]
Hi. Following that great answer from Mark.
Just to tell that yes it can totally be deployed without HA. You can either use installation methods 1 & 2, see here: [https://emhass.readthedocs.io/en/latest/intro.html#configuration-and-installation](https://emhass.readthedocs.io/en/latest/intro.html#configuration-and-installation)
The dockerized solution makes thing easier.
However, yes the HA integration makes everything easier, adding to the power of HA to manage entities, database, automations, etc. Along with the added plus of a plethora of other integrations available in HA.

**Reply by davidusb-geek** [reply]
The only big limitation of using EMHASS without HA is that you will need to pass all the data that it needs at runtime. Namely the load forecast needed for the algorithm to work properly: 
- Load power forecast: how much power your house will demand on the next 24h. This is given in Watts.

**Comment by rthorntn** [comment]
Thanks David, sounds like implementing HASS makes sense.

I already hold my home energy data in node-red global variables, any idea if I can feed that into a standalone EMHASS, or can EMHASS read the values directly through MQTT?

Is the "dashboard" image Mark shared in his reply from HASS or EMHASS?

I need to go deep into EMHASS, any good videos out there?

Also, which forum is best to get EMHASS help?

**Reply by purcell-lab** [reply]
> I already hold my home energy data in node-red global variables, any idea if I can feed that into a standalone EMHASS, or can EMHASS read the values directly through MQTT?

The interface with EMHASS is via REST POST commands, some users are doing this directly in node red.  So there is nothing stopping you exporting your forecasts via REST direct from node red into the EMHASS.

My dashboard is from Home Assistant.  

EMHASS exposes a web service dashboard which can be used as a standalone capability:

![image](https://github.com/davidusb-geek/emhass/assets/79175134/059e1473-4a1b-4920-baeb-0fc7a6c2e90d)
![image](https://github.com/davidusb-geek/emhass/assets/79175134/12903e6a-3119-41a5-b607-8554edd388db)

**Comment by rthorntn** [comment]
Hi Again!

So I haven't used Docker before but I've been using Linux for the best part of 20 years.

So to have a play (without HASS) so I've downloaded Docker Desktop (Mac) and pulled the image, it just errors:

```
web_server INFO  Failed to find secrets_emhass.yaml in directory:/app/secrets_emhass.yaml
2024-06-18 16:48:40 web_server INFO  Attempting to use secrets from arguments or environment variables
2024-06-18 16:48:40 web_server ERROR No specified Home Assistant URL
2024-06-18 16:48:40 Traceback (most recent call last):
2024-06-18 16:48:40   File "<frozen runpy>", line 198, in _run_module_as_main
2024-06-18 16:48:40   File "<frozen runpy>", line 88, in _run_code
2024-06-18 16:48:40   File "/usr/local/lib/python3.11/dist-packages/emhass/web_server.py", line 411, in <module>
2024-06-18 16:48:40     raise Exception("Can not find Home Assistant URL from secrets_emhass.yaml or ARG/ENV") 
2024-06-18 16:48:40     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-06-18 16:48:40 Exception: Can not find Home Assistant URL from secrets_emhass.yaml or ARG/ENV
```

I've had a look at the documentation, tbh its confusing me, it all seems to be about deferred loads, I can't really see anything about my use case or deployments without HASS.  I understand that I need to feed it the load power forecast somehow, my forecast is roughly the same every day (~14kw) as I only have one consistent large load at 4kwh (1kw over 4 hours, pool pump).

I'm familiar with Postman and the like for RestAPI.  Node-Red can use RestAPI.

My use case is as previously stated, I have no need to defer loads at this time.

To start with I essentially want to provide a day time target SoC (~100%) to be reached by a sunset, same with a night time target SoC (~30%) by sunrise.  The day time one is so the battery is full during the day to sell in the evening shoulder and the night time one is so I can stay on battery and have battery power for the morning shoulder.  I don't want to be stung topping up my battery during power spikes.

As mentioned I have wholesale 5 minute billing, all my data is in MQTT and I process it in node-red, store it in influxdb and display it with grafana.

Is EMHASS even able to provide instructions in 5 minute chunks that can be used to tell my battery to import or export a certain amount of power (import max 3kwh, export max 5kwh)...

The output I need is whether in this 5 minute window I should do nothing, sell battery or buy grid (and how many watts).

Inputs:

SoC
Cost
Solar watts
Load watts
Grid watts
Battery watts

If I need to use grid power to charge my battery I'd really want to do that when its cheap.
If I sell battery power I'd really like to sell it when its expensive.

So for EMHASS to tell me when to:

Buy cheap power to maintain SoC and give me power to sell in spikes
Sell expensive power to make me money

Sorry to be a pain, I just can't really see anything in the docs for this use case.

My setup:
7.4kw panels
19.2kwh battery
MPP solar to battery to house load inverter (MQTT)
Growatt solar to grid inverter (MQTT)
GoodWe battery to grid inverter (MODBUS TCP)
Victron smartshunt (MQTT)
Shelly 3EM (MQTT)

Please help Obi Wan.

Thanks!
Richard
